Ginger/Ginger/UserControlsLib/UCDataMapping.xaml.cs

```diff
@@ -0,0 +1,553 @@
+            else if(variabelsSourceList != null)
+            {
+                ucDataMapping.SetValue(UCDataMapping.VariabelsSourceProperty, variabelsSourceList);
+            }
+
+            if (string.IsNullOrEmpty(outputVariabelsSourceProperty) == false)
+            {
+                Binding outputVariabelsSourceBinding = new Binding(outputVariabelsSourceProperty);
+                outputVariabelsSourceBinding.Mode = BindingMode.OneWay;
+                outputVariabelsSourceBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+                ucDataMapping.SetBinding(UCDataMapping.OutputVariabelsSourceProperty, outputVariabelsSourceBinding);
+            }
+
+            Binding mappedItemTypeBinding = new Binding(dataTypeProperty);
+            mappedItemTypeBinding.Mode = BindingMode.TwoWay;
+            mappedItemTypeBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+            ucDataMapping.SetBinding(UCDataMapping.MappedTypeProperty, mappedItemTypeBinding);
+
+            Binding mappedValueBinding = new Binding(dataValueProperty);
+            mappedValueBinding.Mode = BindingMode.TwoWay;
+            mappedValueBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+            ucDataMapping.SetBinding(UCDataMapping.MappedValueProperty, mappedValueBinding);
+
+            if (string.IsNullOrEmpty(enableDataMappingProperty) == false)
+            {
+                Binding allowDataMappingBinding = new Binding(enableDataMappingProperty);
+                allowDataMappingBinding.Mode = BindingMode.OneWay;
+                allowDataMappingBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+                ucDataMapping.SetBinding(UCDataMapping.EnableDataMappingProperty, allowDataMappingBinding);
+            }
+            else
+            {
+                ucDataMapping.SetValue(UCDataMapping.EnableDataMappingProperty, true);
+            }
+
+            template.VisualTree = ucDataMapping;
+            return template;
+        }       
+        #endregion Global
+
+        #region Variables
+        private static void OnVariabelsSourcePropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            var control = sender as UCDataMapping;
+            if (control != null)
+            {
+                control.VariabelsSourcePropertyChanged((ObservableList<string>)args.NewValue);
+            }
+        }
+        private void VariabelsSourcePropertyChanged(ObservableList<string> variabelsSourceList)
+        {
+            if (variabelsSourceList == null || variabelsSourceList.Count == 0 || (variabelsSourceList.Count == 1 && variabelsSourceList[0] == string.Empty))
+            {
+                GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.Variable);
+            }
+            if (variabelsSourceList != null && variabelsSourceList.Where(x=>string.IsNullOrEmpty(x)==false).ToList().Count > 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)